modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java

```diff
@@ -535,12 +538,54 @@ public void handle(HttpExchange exchange) throws IOException {
         }
     }
 
+    public void testReadDoesNotRetryForRepositoryAnalysis() {
+        final int maxRetries = between(0, 5);
+        final int bufferSizeBytes = scaledRandomIntBetween(
+            0,
+            randomFrom(1000, Math.toIntExact(S3Repository.BUFFER_SIZE_SETTING.get(Settings.EMPTY).getBytes()))
+        );
+        final BlobContainer blobContainer = createBlobContainer(maxRetries, null, true, ByteSizeValue.ofBytes(bufferSizeBytes));
+
+        final byte[] bytes = randomBlobContent();
+
+        @SuppressForbidden(reason = "use a http server")
+        class FlakyReadHandler implements HttpHandler {
+            private int failureCount;
+
+            @Override
+            public void handle(HttpExchange exchange) throws IOException {
+                if (failureCount != 0) {
+                    ExceptionsHelper.maybeDieOnAnotherThread(new AssertionError("failureCount=" + failureCount));
+                }
+                failureCount += 1;
+                Streams.readFully(exchange.getRequestBody());
+                final var bytesSent = sendIncompleteContent(exchange, bytes);
+                assertThat(bytesSent, greaterThan(0));
```

Comment:
It seems to me that `bytesSend` can be `0`?

Reply:
Ah yes, my mistake, it can indeed send 0 bytes.